Python/Product/Analyzer/Intellisense/OutOfProcProjectAnalyzer.cs

```diff
@@ -993,9 +993,11 @@ class ImportStatementWalker : PythonWalker {
             }
 
             public override bool Walk(FromImportStatement node) {
-                var name = node.Root.MakeString();
-                if (!_analyzer.IsModuleResolved(_entry, name, node.ForceAbsolute)) {
-                    Imports.Add(MakeUnresolvedImport(name, node.Root));
+                var names = ModuleResolver.GetModuleNamesFromImport(_entry, node);
+                foreach (var n in names) {
+                    if (!_analyzer.IsModuleResolved(_entry, n, node.ForceAbsolute)) {
+                        Imports.Add(MakeUnresolvedImport(n, node.Root));
+                    }
```

Comment:
We should only add the unresolved import if none of the names are resolved. Otherwise we will warn everywhere that we generate multiple names for the same module, even when the "obvious" one is there.

Reply:
This is to handle `from . import a, b, c` where only `a` is actually resolved. However, it can be limited to `.` case since in other cases it may not be necessary.

Reply:
So I guess we can add all names if import is from `.` or similar and keep old approach for other cases?

Reply:
Actually, implementation of `IsModuleResolved` has changed - it no longer does TryImport so it won't report 'unresolved' entries. It only reports those that were not found at all and reported via 'unresolved' in TryImportModules. So here it only adds those names that appears in `GetModuleInfo()?.GetAllUnresolvedModules()`

Reply:
In any case, this is going to change fairly soon to report via diagnostics rather than being a request from VS.